GUI/Utils/DarkModeForms/DarkModeCS.cs

```diff
@@ -0,0 +1,1247 @@
+
+        [LibraryImport("user32")]
+        [DefaultDllImportSearchPaths(DllImportSearchPath.UserDirectories)]
+        private static partial IntPtr GetDC(IntPtr hwnd);
+
+        [LibraryImport("user32")]
+        [DefaultDllImportSearchPaths(DllImportSearchPath.UserDirectories)]
+        private static partial IntPtr ReleaseDC(IntPtr hwnd, IntPtr hdc);
+
+        public static IntPtr GetHeaderControl(ListView list)
+        {
+            const int LVM_GETHEADER = 0x1000 + 31;
+            return SendMessage(list.Handle, LVM_GETHEADER, IntPtr.Zero, "");
+        }
+
+        #endregion
+
+        /// <summary>'true' if Dark Mode Color is set in Windows's Settings.</summary>
+        public bool IsDarkMode { get; set; }
+
+        /// <summary>Windows Colors. Can be customized.</summary>
+        public OSThemeColors OScolors { get; set; }
+
+        /// <summary>Constructor.</summary>
+        public DarkModeCS(bool debugTheme = false)
+        {
+            DebugTheme = debugTheme;
+            SystemEvents.UserPreferenceChanged += new UserPreferenceChangedEventHandler(OnUserPreferenceChanged);
+        }
+
+        #region Public Methods
+
+        /// <summary>This tries to style and automatically apply Windows Dark Mode (if enabled) to a Form.</summary>
+        /// <param name="_Form">The Form to become Dark</param>
+        public void Style(Form _Form)
+        {
+            ApplyTheme(_Form);
+            if (!StyledForms.Contains(_Form))
+            {
+                StyledForms.Add(_Form);
+            }
+        }
+
+        /// <summary>Recursively apply the Colors from 'OScolors' to the Control and all its childs.</summary>
+        /// <param name="control">Can be a Form or any Winforms Control.</param>
+        public void ThemeControl(Control control)
+        {
+            BorderStyle BStyle = BorderStyle.FixedSingle;
+            FlatStyle FStyle = FlatStyle.Flat;
+
+            //Change the Colors only if its the default ones, this allows the user to set own colors:
+            if (control.BackColor == SystemColors.Control || control.BackColor == SystemColors.Window)
+            {
+                control.GetType().GetProperty("BackColor")?.SetValue(control, OScolors.Control);
+            }
+            if (control.ForeColor == SystemColors.ControlText || control.ForeColor == SystemColors.WindowText)
+            {
+                control.GetType().GetProperty("ForeColor")?.SetValue(control, OScolors.TextActive);
+            }
+
+            var borderStyleInfo = control.GetType().GetProperty("BorderStyle");
+            if (borderStyleInfo != null)
+            {
+                BorderStyle borderStyle = (BorderStyle)borderStyleInfo.GetValue(control);
+                if ((BorderStyle)borderStyle != BorderStyle.None)
+                {
+                    borderStyleInfo.SetValue(control, BStyle);
+                }
+            }
+
+            control.HandleCreated += (object sender, EventArgs e) =>
```

Comment:
These events will unintentionally keep these controls alive, they need to be unbound at some point